### PR TITLE
Inject EventDispatcher into ControllerAbstract

### LINKED ISSUE #266: Inject the EventDispatcher to ControllerAbstract constructor
It should replace the [template](https://github.com/fsi-open/admin-bundle/blob/master/Controller/ControllerAbstract.php#L53) parameter. This will remove the need for the `setEventDispatcher` method and the relevant compiler pass.

### PR DESCRIPTION
**Pull request by szymach**
Resolves #266

- [x] Update changelog

### CODE REVIEW

**Comment by szymach** [comment]
@fsi-open

File: Controller/ControllerAbstract.php [hunk 1]
```diff
@@ -45,24 +46,18 @@
     /**
      * @param \Symfony\Bundle\FrameworkBundle\Templating\EngineInterface $templating
      * @param \FSi\Bundle\AdminBundle\Admin\Context\ContextManager $contextManager
-     * @param string|null $resourceActionTemplate
```
Comment (chives) [inline Controller/ControllerAbstract.php]: docblock is now inconsistent with real signature

**Comment by chives** [comment]
`SetEventDispatcherPass` and `ControllerAbstract::setEventDispatcher()` should also be marked as deprecated in `2.1` branch

**Comment by szymach** [comment]
@fsi-open

**Comment by szymach** [comment]
@fsi-open

**Comment by szymach** [comment]
@rn0 @JarekW

**Pull request merged**